content/Stop creating services for everything - Knowing YAAF & the FormObject Pattern.md

```diff
@@ -0,0 +1,130 @@
+  end
+end
+```
+
+It looks terrible, doesn't it? Maybe our first thinking about that piece of code is to make a refactoring moving the creation of the post to a service named `PostCreationService`, which could be useful and it might be used in the future in another part of the system. But... what have we said about _reinventing the wheel_?
+
+Using **YAAF** we should create a new `PostForm` class that is going to encapsulate all the logic of post creation and related models inside it. And is very simple to implement it! Look at this code:
+```ruby
+# app/forms/registration_form.rb
+
+class PostForm < ApplicationForm
+  attr_accessor :post_form_params
+
+  def initialize(args = {})
+    @post_form_params = args
+    @models = [post, category, tags].flatten.compact
+  end
+
+  def post
+    @post ||= Post.new(post_form_params[:post])
+  end
+
+  def category
+    return [] if post_form_params[:category_name].blank?
+
+    post.category = Category.find_or_initialize_by(name: post_form_params[:category_name])
+    post.category
+  end
+
+  def tags
+    return [] if post_form_params[:tags].blank?
+
+    post.tags = post_form_params[:tags].map do |tag|
+      tag[:id].present? ? Tag.find(tag[:id]) : Tag.find_or_initialize_by(name: tag[:name])
+    end
+    post.tags
+  end
+end
+```
+
+And then in our controller
+
+```ruby
+class Api::V1::PostsController < Api::V1::ApiController
+  def index
+    @posts = Post.all
+  end
+
+  def create
+    form = PostForm.new(post_form_params)
+    form.save!
+    @post = form.post
+  end
+
+  private
+
+  def post_form_params
+    params.permit(:category_name, tags: %i[id name], post: %i[title body publisher_id category_id])
+  end
+end
+```
+
+**Tip:** Having an `ApplicationForm` which inherits from `YAAF::Form` is a good practice.
+
+That's it, now we have a `PostForm` which encapsulates all the persistency logic of post/tags/categories, leaving our controller and models clean and the code is easy to follow.
+
+Another good thing is that **YAAF** provides a similar API to `ActiveModel` models so you can treat them interchangeably.
+
+## Why not a Service or PORO's?
+
+- **Make customized Services or PORO's could be disorganized** if you're working in a team.
+- YAAF helps you to apply the **Form Pattern** in an easy way.
+- YAAF is only **64 lines long**.
+- Is **well tested and maintained**.
+- It helps you keep your models, views and controllers thin by providing a better place where to put business logic. In the end, this will **improve the quality of your codebase and make it easier to maintain and extend**.
+- And a lot **[more](https://github.com/rootstrap/yaaf)**.
+
+## Summary
+
+Well if you come along this way I hope this article helps you to integrate **YAAF** in your project and start using the `FormObject` Pattern to make your code even better, you can see more examples **[here](https://yaaf-examples.herokuapp.com/)**. **YAAF** is open-source and is open to receive new contributions so check it out!
```

Comment:
It'd be better to link to the github instead of the example app I think. There are more examples and explanations in the readme than in the demo app.